### DISCUSSION TITLE
Menubar how to set active MenuItem?

### DISCUSSION THREAD

**Discussion by Koishiiko** [primefaces/community#2564] [PrimeVue] [answered]
I am new to use PrimeVue. I want to control the currently highlighted `MenuItem` in the `Menubar` by a variable. 

I checked the documentation, and it seems that there have no API about this. But I found an `active` variable from [here](https://primevue.org/menubar/#api.menubar.interfaces.MenubarContext), so I tried using this variable to implement it([playground](https://stackblitz.com/edit/primevue-4-ts-vite-issue-template-2h9wrp?file=src%2FApp.vue)):

```vue
<template>
  <Menubar :model="items" :pt="pt" />
</template>

<script setup>
const items = ref([
  { key: 'Home', label: 'Home' },
  { key: 'Features', label: 'Features' },
  { key: 'Projects', label: 'Projects' },
  { key: 'Contact', label: 'Contact' },
]);

const pt = {
  itemContent: (options) => options.context.active ? 'active': null,
};
</script>

<style scoped>
:deep(.active) {
  background: grey !important;
}
</style>
```

But it looks like that `options.context.active` is always `false`, so it maybe not working. (or am I using it incorrectly?)

Is there have any solution that can get(or set) the active `MenuItem`?  I think it should have some method like this:

```vue
<template>
  <Menubar :model="items" v-model:active="activeItemKey" />
</template>

<script setup>
const activeItemKey = ref('Home');
</script>
```

**Comment by j0rgedev** [comment, marked as answer]
Hello @Koishiiko. Sakai Template has an example of how that could be approached. Please refer to this file: [Sakai Template - AppMenuItem](https://github.com/primefaces/sakai-vue/blob/master/src/layout/AppMenuItem.vue)

**Reply by Koishiiko** [reply]
Thanks a lot! I think I will use this solution from template. By the way, I noticed that `Sakai Template` dose not use PrimeVue's `Menu/Menubar` component but insted reimplements it. Is it mean that using the existing components to achieve it is more difficult than defining a custom component to do so? I tried some other approaches using `Menubar`'s API, but they were all rather complex and redundant.

**Reply by j0rgedev** [reply]
That's a good point.  I haven't used those Menu/Menubar Components either. I guess sakai vue didn't implement it because, as you said, menubar's API could fit better in other scenarios and they just wanted to make it simpler. By the way, I plan on forking sakai vue and doing a sakai plus version with a better sidebar. I could try using menubar component, and I'll reach out to you 👍🏻
